include/veriblock/storage/block_repository_rocks.hpp

```diff
@@ -5,13 +5,107 @@
+#include "veriblock/blob.hpp"
 #include "veriblock/serde.hpp"
 #include "veriblock/storage/block_repository.hpp"
 #include "veriblock/storage/db_error.hpp"
 #include "veriblock/strutil.hpp"
 
 namespace VeriBlock {
 
+//! column family type
+using cf_handle_t = rocksdb::ColumnFamilyHandle;
+
+template <typename Block>
+struct CursorRocks : public Cursor<typename Block::hash_t, Block> {
+  //! stored block type
+  using stored_block_t = Block;
+  //! block has type
+  using hash_t = typename Block::hash_t;
+
+  CursorRocks(std::shared_ptr<rocksdb::DB> db,
+              std::shared_ptr<cf_handle_t> hashBlockHandle)
+      : _db(db), _hashBlockHandle(hashBlockHandle) {
+    auto iterator = _db->NewIterator(rocksdb::ReadOptions(), _hashBlockHandle.get());
+    _iterator = std::unique_ptr<rocksdb::Iterator>(iterator);
+  }
+
+  void seekToFirst() override { _iterator->SeekToFirst(); }
+  void seek(const hash_t& key) override {
+    std::string blockHash(reinterpret_cast<const char*>(key.data()),
+                          key.size());
+    _iterator->Seek(blockHash);
+  }
+  void seekToLast() override { _iterator->SeekToLast(); }
+  bool isValid() const override { return _iterator->Valid(); }
+  void next() override { _iterator->Next(); }
+  void prev() override { _iterator->Prev(); }
+
+  hash_t key() const override {
+    auto key = _iterator->key();
+    return hash_t(key.ToString());
+  }
+
+  stored_block_t value() const override {
+    auto value = _iterator->value();
+    return stored_block_t::fromRaw(value.ToString());
+  }
+
+ private:
+  std::shared_ptr<rocksdb::DB> _db;
+  std::shared_ptr<cf_handle_t> _hashBlockHandle;
+  std::unique_ptr<rocksdb::Iterator> _iterator;
+};
+
+template <typename Block>
+struct WriteBatchRocks : public WriteBatch<Block> {
+  //! stored block type
+  using stored_block_t = Block;
+  //! block has type
+  using hash_t = typename Block::hash_t;
+  //! block height type
+  using height_t = typename Block::height_t;
+
+  WriteBatchRocks(std::shared_ptr<rocksdb::DB> db,
+                  std::shared_ptr<cf_handle_t> hashBlockHandle)
+      : _db(db), _hashBlockHandle(hashBlockHandle) {}
+
+  void put(const stored_block_t& block) override {
+    std::string blockHash(reinterpret_cast<const char*>(block.hash.data()),
+                          block.hash.size());
+    std::string blockBytes = block.toRaw();
+    rocksdb::Status s = _batch.Put(_hashBlockHandle.get(), blockHash, blockBytes);
+    if (!s.ok() && !s.IsNotFound()) {
+      throw db::DbError(s.ToString());
+    }
+  }
+
+  void removeByHash(const hash_t& hash) override {
+    std::string blockHash(reinterpret_cast<const char*>(hash.data()),
+                          hash.size());
+    rocksdb::Status s = _batch.Delete(_hashBlockHandle.get(), blockHash);
+    if (!s.ok() && !s.IsNotFound()) {
+      throw db::DbError(s.ToString());
+    }
+  }
+
+  void clear() override { _batch.Clear(); }
+
+  void commit(BlockRepository<Block>& /* ignore */) override {
```

Comment:
why ignore?

Reply:
We do not have any way to access DB instance from the BlockRepository interface. Only way is to pass DB ref during WriteBatch creation. Therefore we do not need BlockRepository to commit batch to the DB.

Reply:
maybe remove argument?

Reply:
We will have to change the interface as well. So BlockRepositoryInmem will be changed accordingly. Do you want to do it?

Reply:
If it is easy to do, yeah